arango/resolver.py

```diff
@@ -13,51 +13,54 @@
 class HostResolver(ABC):  # pragma: no cover
     """Abstract base class for host resolvers."""
 
-    def __init__(self) -> None:
-        self._max_tries: int = 3
+    def __init__(self, host_count: int = 1, max_tries: int = 3) -> None:
+        if max_tries < host_count:
+            raise ValueError("max_tries cannot be less than host_count")
+
+        self._host_count = host_count
+        self._max_tries = max_tries
 
     @abstractmethod
-    def get_host_index(self, prev_host_index: Optional[int] = None) -> int:
+    def get_host_index(self, indexes_to_filter: Set[int] = set()) -> int:
```

Comment:
Using a mutable object like a set as default arguments should be avoided ([reference](https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments)). I recommend using `None` here instead and initialize the set in the function.